openedx/core/djangoapps/user_api/preferences/tests/test_views.py

```diff
@@ -168,7 +167,7 @@ def test_create_preferences_other_user(self, api_client, user):
                 "dict_pref": {"int_key": 10},
                 "string_pref": "value",
             },
-            expected_status=403 if user == "staff_user" else 404,
```

Comment:
Why is this changing?  Was this just always a bad idea?

Reply:
Exactly.  This was always a bad idea.  😄 

If a user doesn't have access, they should just get a 403.
To prevent leakage of the existence of a user, just return a 403 if you don't have access.